### PR TITLE
Add the message header for socket transport

### LINKED ISSUE #23: Bug in socket client.
@rockers7414  I set the socket client buffer to 2048, but only get the length 1460 of response's data. Besides, Isn't it socket stream? why I still need to set the client receive buffer?
`response = str(sock.recv(1024), "utf-8")`

[client_console.txt](https://github.com/rockers7414/xmusic/files/830199/client_console.txt)
[xmusic-daemon.txt](https://github.com/rockers7414/xmusic/files/830200/xmusic-daemon.txt)

### PR DESCRIPTION
**Pull request by rockers7414**
This pull request fixes #23

### CODE REVIEW

File: daemon/xmusic-cli.py [hunk 1]
```diff
@@ -22,8 +23,16 @@
 
     with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
         sock.connect(server_addr)
-        sock.sendall(bytes(request + "\n", "utf-8"))
-        response = str(sock.recv(1024), "utf-8")
+        header = pack("=BHB", 29, len(request), 29)
```
Comment (cclien0725) [inline daemon/xmusic-cli.py]: The `header` may use `MessageHeader` to create.

File: daemon/xmusic-cli.py [hunk 2]
```diff
@@ -22,8 +23,16 @@
...
+        sock.sendall(header + bytes(request, "UTF-8"))
+
+        length = unpack("=BHB", sock.recv(4))[1]
```
Comment (cclien0725) [inline daemon/xmusic-cli.py]: The `length` may use `MessageHeader` to get the data length.

File: daemon/xmusic-cli.py [hunk 3]
```diff
@@ -22,8 +23,16 @@
...
+        response = ""
+        while len(response) < length:
+            buf = sock.recv(min(1024, length - len(response)))
```
Comment (cclien0725) [inline daemon/xmusic-cli.py]: The value of `1024` may use a constant to unified management. ex. `RPCHandler. MAX_BUF_LENGTH`.

**Review by cclien0725 — CHANGES_REQUESTED** [review]
There are some suggestions for you. Please revised it. Thanks.

**Comment by rockers7414** [comment]
Sorry, I just don't mention that the CLI should be refactored by @KocaMio. ;)

**Comment by rockers7414** [comment]
Okay, I've already completed the CLI part. Below is the sample of the result. Please help to review.
```
➜  daemon git:(add-msg-header) ./xmusic-cli.py
API loading...
****************************************
Welcome to xmusic-cli.
****************************************
xmusic> get_artist --artist_name "Ed Sheeran"
{"method": "get_artist", "params": {"artist_name": "Ed Sheeran", "source": "db"}, "jsonrpc": "2.0", "id": "e2476551-66cd-48c7-a643-17f066bd961e"}
{
    "result": [
        {
            "provider_id": "8e9d7910-0376-4c2d-8a43-264ec58b35e9",
            "name": "Ed Sheeran",
            "artist_id": "20f4ed21-666a-45ff-b530-e0049854e2d6",
            "provider_res_id": "6eUKZXaKkcviH0Ku9w2n3V",
            "popularity": "100",
            "images": [
                {
                    "height": "160",
                    "image_id": "95b1a015-568e-4f30-aea0-75892c086e7a",
                    "path": "https://i.scdn.co/image/f39b705ba6b1070c77bce4e9c21dff5bac96f21c",
                    "width": "160"
                },
                {
                    "height": "320",
                    "image_id": "5675803f-a562-4ba4-bb1e-1c950b80e4e4",
                    "path": "https://i.scdn.co/image/e21f791b600f04f9dd8ad2b6d9b3304a82e048ca",
                    "width": "320"
                },
                {
                    "height": "640",
                    "image_id": "3905342b-d08e-4fba-9ba1-53038f6e2729",
                    "path": "https://i.scdn.co/image/f0370da3f52161b07a461b4be9a64d0adbfb498d",
                    "width": "640"
                }
            ],
            "genres": [
                {
                    "genre_id": "7088b0fd-8ff7-49bf-b374-08280afb4f93",
                    "name": "pop"
                }
            ]
        }
    ],
    "id": "e2476551-66cd-48c7-a643-17f066bd961e",
    "jsonrpc": "2.0"
}
xmusic>
```

**Review by cclien0725 — APPROVED** [review]
Looks great. Nice job!

**Comment by rockers7414** [comment]
Thanks, I'll merge the pull request tomorrow night if no one has any concern.

**Pull request merged**